src/Text/Pandoc/Writers/Markdown/Inline.hs

```diff
@@ -95,6 +96,13 @@ escapeText opts = T.pack . go' . T.unpack
                   , isAlphaNum x -> c : '_' : x : go xs
                 _                -> c : go cs
 
+-- Escape the escape character, as well as formatting pairs
+escapeMarkuaString :: Text -> Text
+escapeMarkuaString s = foldr (uncurry T.replace) s $ zip ["--", "**", "//",
+                                                          "^^", ",,"]
+                                                          ["~-~-", "~*~*",
+                                                           "~/~/", "~^~^",
+                                                           "~,~,", "~~"]
```

Comment:
I don't think it's a good idea to use `zip` in this way -- better just to use an explicit list, where each item is paired with its escaped counterpart. (Think what would happen if one item got deleted from just one of these lists, everything would pair up with the wrong counterpart.)

Reply:
We changed it accordingly